src/index.js

```diff
@@ -38,29 +39,35 @@ function buildValidations(referenced, dereferenced, receivedOptions) {
     const options = getOptions(receivedOptions);
 
     const schemas = {};
-    Object.keys(dereferenced.paths).forEach(function (currentPath) {
-        const parsedPath = dereferenced.basePath && dereferenced.basePath !== '/'
-            ? dereferenced.basePath.concat(currentPath.replace(/{/g, ':').replace(/}/g, ''))
-            : currentPath.replace(/{/g, ':').replace(/}/g, '');
-        schemas[parsedPath] = {};
-        Object.keys(dereferenced.paths[currentPath])
-            .filter(function (parameter) { return parameter !== 'parameters' })
-            .forEach(function (currentMethod) {
-                const parsedMethod = currentMethod.toLowerCase();
-
-                let requestValidator;
-                if (options.buildRequests) {
-                    requestValidator = buildRequestValidator(referenced, dereferenced, currentPath,
-                        parsedPath, currentMethod, options);
-                }
 
-                let responseValidator;
-                if (options.buildResponses) {
-                    responseValidator = buildResponseValidator(referenced, dereferenced, currentPath, parsedPath, currentMethod, options);
-                }
+    const basePaths = dereferenced.servers
+        ? dereferenced.servers.map(({ url }) => URL.parse(url).pathname)
+        : [dereferenced.basePath || '/'];
 
-                schemas[parsedPath][parsedMethod] = Object.assign({}, requestValidator, { responses: responseValidator });
-            });
+    Object.keys(dereferenced.paths).forEach(function (currentPath) {
+        basePaths.forEach(basePath => {
```

Comment:
Since this is currently always applied, wouldn't that be a breaking change? Wasn't the idea to make this behaviour configurable?

Reply:
I don't think it's a breaking change. In the current version if you use OAS3 and have `servers` configured with a base path then no validation is done since the request paths never match the schema paths. This fixes that scenario without affecting cases where you have no base path specified, or you're using OAS2.

Reply:
@tamlyn IMO, actually this what makes this change a breaking one.

As endpoints that previously had no validation could fail now after this change.

Don't you think so?

Reply:
Oh, I think I understand.

 I was seeing things from the perspective of `openapi-validator-middleware` where the current behaviour is that it does nothing at all – no validation. This fix would cause validation to be run which could start causing endpoints to fail if they were being called with invalid data. But of course, that's what you want since this is a validator.

However from the perspective of `api-schema-builder` it's possible that someone else is consuming it and already working around the lack of base path, and this fix would break that.

Reply:
@tamlyn Even from a perspective of `openapi-validator-middleware` - imagine that I have a system in production that has inaccurate OpenAPI spec that is mismatched with actual requests coming in, but it works, because currently endpoint is skipped due to baseUrl missing. I bump version of `openapi-validator-middleware`, and bam, my production is broken, requests that worked yesterday work no more. Do I want to fix my OpenAPI spec eventually? Absolutely, but I don't expect things to break like that unless I did a major version upgrade.